reqwest-tracing/src/reqwest_otel_span_macro.rs

```diff
@@ -127,6 +127,7 @@ macro_rules! reqwest_otel_span {
             let header_default = &::http::HeaderValue::from_static("");
             let user_agent = format!("{:?}", $request.headers().get("user-agent").unwrap_or(header_default)).replace('"', "");
 
+            #[cfg(not(feature = "deprecated_fields"))]
             macro_rules! request_span {
```

Comment:
I'm not sure if this will compile but I'm imagining something more like this. That way we're publishing both fields when `deprecated_fields` is enabled

```rust
#[cfg(feature = "deprecated_fields")]
macro_rules! request_span {
    ($lvl:expr) => {
        $crate::reqwest_otel_span_macro::private::span!(
            $lvl,
            "HTTP request",
            http.request.method = %method,
            #[cfg(feature = "deprecated_fields")]
            http.method = %method,
            $($field)*
        )
    }
}
```

Reply:
Thanks Ethan, for all the comments.

It is not possible to use conditional compilations inside the span! macro. I tried several workarounds, and in the end, the easiest solution is to duplicate the macro. Use #[cfg(not(feature = "deprecated_attributes"))] for the new attributes and #[cfg(feature = "deprecated_attributes")] for both, the old and new attributes.

This should compile correctly; I have seen other [crates] (https://github.com/iliekturtles/uom/blob/5dca5a69330fd186c7f840548f6b959591574193/src/features.rs#L1-L16) using a similar strategy.